_posts/2021-01-01-getting-started-with-graphql.md

```diff
@@ -0,0 +1,288 @@
+
+We will now add resolvers for all the types defined in the schema. We first add a resolver class named `QueryResolver` containing the methods corresponding to our GraphQL operations: 
+
+```java
+@Service
+public class QueryResolver implements GraphQLQueryResolver {
+
+  private ProductRepository productRepository;
+  
+  @Autowired
+  public QueryResolver(final ProductRepository productRepository) {
+    super();
+    this.productRepository = productRepository;
+  }
+
+  public List<Product> getMyRecentPurchases(Integer count, String customerID) {
+    
+    List<Product> products = productRepository.getRecentPurchases(count);
+    
+    return products;
+  }
+  
+  public List<Product> getLastVisitedProducts(final Integer count, final String customerID) {
+    List<Product> products = productRepository.getLastVisitedPurchases(count);
+    return products;
+  }
+  
+  public List<Product> getProductsByCategory( final String category) {
+    List<Product> products = productRepository.getProductsByCategory(category);
+    return products;
+  }
+
+}
+```
+We have defined the `QueryResolver` class as a Service class to resolve the root Query in our GraphQL schema. This service class is injected with a repository class to fetch data from an H2 Database.
+
+We next add a resolver for the `Manufacturer` object type:
+```java
+@Service
+public class ProductResolver implements GraphQLResolver<Product>{
+
+  private ManufacturerRepository manufacturerRepository;
+  
+  @Autowired
+  public ProductResolver(ManufacturerRepository manufacturerRepository) {
+    super();
+    this.manufacturerRepository = manufacturerRepository;
+  }
+
+  public Manufacturer getMadeBy(final Product product) {
+    return manufacturerRepository.getManufacturerById(product.getManufacturerID());
+  }
+}
+```
+This resolver fetches the `Manufacturer` record corresponding to a `Product`.
+
+### Connecting to Datasources and Applying Middleware Logic
+Next, we will enable our resolvers to fetch data from underlying data sources like a database or web service. For this example, we have configured an in-memory H2 database as the data store for `products` and `manufacturers`. We use Spring JDBC to retrieve data from the database and put this logic in separate repository classes. 
+
+Apart from fetching data, we can also build different categories of middleware logic in this business service layer, like authorization of incoming requests, applying filters on data fetched from backend, transformation into backend data models, and also caching any less frequently changing data.
+
+### Running Application
+After compiling and running the application, we can send GraphQL queries to the endpoint http://localhost:8080/graphql. A sample request and response captured in postman tool is shown here :
+
+![Snippet from Postman Tool](/assets/img/posts/graphql-intro/results.png)
```

Comment:
Can you please include the request and response as code snippets in the text instead of as an image? That's more accessible and looks better.